proposed/http-client/http-client.md

```diff
@@ -3,16 +3,27 @@ HTTP Client
 
 This document describes common interfaces for sending HTTP messages.
 
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
 
 ## Specification
 
+### Client
+
 An HTTP client has the responsibility to send a PSR-7 request and return a PSR-7
-response. When there is an error during sending the request or an error with network an
-exception should be thrown. 
+response. The HTTP client MAY modify the response from the server before returning 
+the PSR-7 response. 
```

Comment:
Something like this (minus typo ^^) would be better IMO:

"The HTTP client MAY modify the request, given by the user and sent to the server or the response from the server before returning. In this case the request and the response must be consistent between the body and headers. I.e. a server may return a gzip encoded body and the client may know how to decode this, when it decodes the body the client MUST also remove the header that specify this encoding."

Reply:
if this is what we mean by this, then yes lets give this example to illustrate what we actually mean.

Reply:
Thank you.